### ISSUE TITLE
Add support for build-image job

### ISSUE THREAD

**Issue by jmreicha** [CircleCI-Public/aws-ecr-orb#177]
## What would you like to be added

It would be nice to have a shorthand way of building an image only without having to create a job and do the configuration instead in the workflow. Unless I’m missing something it doesn’t look like this is currently possible.

## Why is this needed

It would be useful (for my use case anyway) to have the ability to condense down build/push jobs into workflows.

**Comment by brivu** [comment]
Hey @jmreicha, 

Can you give me some more details about your use case?  I would like to understand more to see if I can make the changes you need.  

Best, 
Brian

**Comment by jmreicha** [comment]
Sure. It might also not be worth the effort if I'm the only person with this use case 😂 

Basically I have a repo that is dedicated to managing Docker images. As part of the flow, the first step is to open a PR and build an image and do various testing. Once the PR is merged into the main branch then the build and push happens.

So as it works right now, I have a bunch of logic in individual jobs for the first build part of the flow, that is pretty much boilerplate, similar to below.

```
...
orbs:
  aws-ecr: circleci/aws-ecr@7.3.0
commands:
  build-image: aws-ecr/build-image
...
jobs:
  build-image-example:
    executor: aws-ecr/default
    steps:
      - checkout
      - build-image:
          account-url: AWS_ECR_ACCOUNT_URL_ENV_VAR_NAME
          path: example
          region: AWS_REGION
          repo: example
          tag: $CIRCLE_SHA1
```

Then in the workflow section I have the following.

```
docker_build_and_push_example:
    when:
      or:
        - << pipeline.parameters.example-modified >>
    jobs:
      - build-image-example:
          context:
            - aws
          filters:
            branches:
              ignore:
                - master
      - aws-ecr/build-and-push-image:
          context:
            - aws
          account-url: AWS_ECR_ACCOUNT_URL_ENV_VAR_NAME
          create-repo: false
          dockerfile: Dockerfile
          no-output-timeout: 20m
          path: example
          region: AWS_REGION
          repo: example
          skip-when-tags-exist: false
          tag: $CIRCLE_SHA1
          filters:
            branches:
              only:
                - master
```

Hopefully that captures the use case and helps clarify. It is basically cumbersome to have a job and workflow, it seems like this could be condensed down into one part of the config, as is I have to jump around the file, which is around 500 lines of yaml.

And I may also be thinking about this the wrong way, in which case I'd be happy to rethink what I'm doing to begin with.

**Comment by brivu** [comment]
Hey @jmreicha, 

[In version 7.3 of this orb, we had a `build-and push-image` command](https://circleci.com/developer/orbs/orb/circleci/aws-ecr?version=7.3.0#commands-build-and-push-image).

Would this work better for you? Other users are also saying the same in issue #178.  

I'm trying to make a case for this next week and your input would help!

Best, 
Brian

**Comment by jmreicha** [comment]
I’m actually more interested in a build-push job for my use case. The reason is basically that it would boil my configs down into one section in the configuration file.

**Comment by brivu** [comment]
Hey @jmreicha, 

In version 8.0 of this orb, there's a `push` parameter that you can set to `false` in the `build-and-push` command.  That way, the image only gets built and no pushing is involved. 

Would that help your use case here in boiling your configs down into one section? If not, can you help me understand why?

Best, 
Brian

**Comment by jmreicha** [comment]
I don't see an easy way to filter build vs build-push on PRs/pushes to master without involving the additional jobs configuration, unless I'm missing something.

I'd be happy to try it out if you point me to an example config?

**Comment by jmreicha** [comment]
I think I see what you mean. Let me give it a try.

**Comment by brivu** [comment]
@jmreicha, 

Just as a note, you need to use the `@dev:alpha` version if you wanted to use have this image built locally without pushing.  

I've added a fix where if `push` is set to false, the docker command adds a `--load` flag which lets you build the image locally without pushing.

It's a fix for issue #179 

Hope that helps! 

Brian

**Comment by jmreicha** [comment]
It looks like they should work. I am going to wait until the new version is released to test but should be good to close this after testing.
